pkg/migration/runner_test.go

```diff
@@ -2202,3 +2202,109 @@ func TestDropAfterCutover(t *testing.T) {
 	assert.Equal(t, tableCount, 0)
 	assert.NoError(t, m.Close())
 }
+
+func TestDeferCutOver(t *testing.T) {
+	tableName := `deferred_cutover`
+	newName := fmt.Sprintf("_%s_new", tableName)
+
+	testutils.RunSQL(t, fmt.Sprintf(`DROP TABLE IF EXISTS %s`, tableName))
+	table := fmt.Sprintf(`CREATE TABLE %s (id bigint unsigned not null auto_increment, primary key(id))`, tableName)
```

Comment:
E2E test looks good. The thing I'm not sure of (and we should test!):

- Are the binary log coordinates still being advanced while we are block waiting (I would hope so)?
- Does it work correctly with resume from checkpoint?

Reply:
I added some additional tests. One of them checks that the binlog position still advances even while the migration is in stateWaitingOnSentinelTable. Another extends the Checkpoint E2E test to manually create a sentinel table after the first migration exists and then ensures that the new migration resumes from checkpoint and correctly blocks on the sentinel, even though both migrations were started with "DeferCutOver: false".